boskos-acquire/0.1/boskos-acquire.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: boskos-acquire
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "boskos,test"
+spec:
+  description: |
+    The boskos-acquire Task will request a resource of the specified type from the
+    server-url. If successful, it will start a pod that will run the boskosctl heartbeat
+    command. When you are done with the resource, release it with boskos-release.
+  params:
+  - name: server-url
+    description: The URL of the running boskos server
+    default: "http://boskos.test-pods.svc.cluster.local"
+  - name: type
+    description: |
+      The type of resource to request. Resource types are specified in the resource
+      ConfigMap provided to the Boskos server.
+    default: gke-project
+  - name: owner-name
+    description: A string that identifies the owner of the leased resource to request.
+  results:
+  - name: leased-resource
+    description: The name of the leased resource
+  steps:
+  - name: boskosctl-acquire
+    image: gcr.io/k8s-staging-boskos/boskosctl@sha256:a7fc984732c5dd0b4e0fe0a92e2730fa4b6bddecd0f6f6c7c6b5501abe4ab105
+    script: |
+      RESOURCE=$(boskosctl acquire \
+        --server-url=$(params.server-url) \
+        --owner-name=$(params.owner-name) \
+        --type=$(params.type) \
+        --state=free \
+        --target-state=busy)
+      echo $RESOURCE > /workspace/full-resource-output.json
+      echo $RESOURCE | jq -rj ".name" > /tekton/results/leased-resource
+  - name: create-heartbeat-pod-yaml
```

Comment:
I am gonna try one last time :stuck_out_tongue_closed_eyes: . I don't see why the step below is worse than the 2 steps proposed :upside_down_face: 

```yaml
  - name: create-heartbeat-pod-yaml
    image: lachlanevenson/k8s-kubectl
    script: |
      FULL_RESOURCE_OUTPUT=$(cat /workspace/full-resource-output.json)
      LEASED_RESOURCE=$(cat /tekton/results/leased-resource)
      cat <<EOF | kubectl apply -f -
      apiVersion: v1
      kind: Pod
      metadata:
        name: boskos-heartbeat-$LEASED_RESOURCE
      spec:
        containers:
        - name: heatbeat
          image: gcr.io/k8s-staging-boskos/boskosctl@sha256:a7fc984732c5dd0b4e0fe0a92e2730fa4b6bddecd0f6f6c7c6b5501abe4ab105
          args:
          - heartbeat
          - --server-url=$(params.server-url)
          - --owner-name=$(params.owner-name)
          - --resource=$FULL_RESOURCE_OUTPUT
          - --period=5m
      EOF
```

You don't need to write that yaml to `/workspace/heartbeat.yaml` ever, it's not gonna be used :upside_down_face:

Reply:
btw, because we are using `:latest` tag on the steps, k8s *will* pull the image each time, so 2 steps is gonna cost more than one :upside_down_face: (if we do use digest or tags for images, that's another story)

Reply:
Okay, seems like you feel strongly enough that I'll combine the two steps into one. I've created https://github.com/tektoncd/catalog/issues/441 to continue the discussion

> btw, because we are using :latest tag on the steps, k8s will pull the image each time, so 2 steps is gonna cost more than one 🙃 (if we do use digest or tags for images, that's another story)

Oh good point, ill add the digest (I think this is a best practice we should recommend for all catalog tasks?)

(also it depends on the image pull policy right?)

Reply:
> (also it depends on the image pull policy right?)

So, I think, whatever the pull policy is, using `:latest` force kubernetes to pull the image each time (because it has *no* way to know which "latest" you wanted :stuck_out_tongue_closed_eyes: )

Reply:
Are you sure?

https://kubernetes.io/docs/concepts/configuration/overview/#container-images
https://kubernetes.io/docs/concepts/containers/images/#updating-images

It sounds like that might only be in the case of having on imagePullPolicy?

> imagePullPolicy is omitted and either the image tag is :latest or it is omitted: Always is applied.

Reply:
Right, with `IfNotPresent` or `Never` it won't try to pull it (if not present for the first one). That said, in the task itself, there is no `imagePullPolicy` so we can "only" assume the default behavior. Overall, the more the catalog grow, the more we should try to follow our recommendation https://github.com/tektoncd/catalog/blob/master/recommendations.md#reference-images-by-digest :upside_down_face: @sthaha validation tool could help there (to warn of tasks that uses latest)